src/components/Button/MuiButton.tsx

```diff
@@ -0,0 +1,123 @@
+import Button, { ButtonProps as MuiBtnProps } from '@mui/material/Button';
+import React, { forwardRef } from 'react';
+
+import { styled } from '@mui/material';
+import Loader from '../Icons/Loader';
+import styles from './Button.module.scss';
+import { tokens } from '../../theme/tokens';
+
+export interface MuiButtonProps extends Omit<MuiBtnProps, 'size'> {
+  loading?: boolean;
+  icon?: React.ReactNode;
+  children?: React.ReactNode;
+  className?: string;
+  size?: 'xs' | 'sm' | 'md' | 'lg' | 'xl';
+  color?: MuiBtnProps['color'] | 'grey' | 'accent';
```

Comment:
maybe extend the colours object so it's reusable everywhere and no need to add `grey | accent` in every component in the future

Reply:
I think it's that ButtonPropsColorOverrides

Reply:
yeah. So the button color works locally due to the typescript overrides done in the muiTheme.d.ts file
but it does not get compiled so when used, the overrides done in the component library does not get recognized